src/workerd/server/workerd-api.c++

```diff
@@ -331,6 +333,185 @@ kj::Maybe<jsg::Bundle::Reader> fetchPyodideBundle(
   return pyConfig.pyodideBundleManager.getPyodideBundle(version);
 }
 
+/**
+ * This function matches the implementation of `getPythonRequirements` in Edgeworker. But it
+ * works on the workerd ModulesSource definition rather than the WorkerBundle.
+ */
+kj::Array<kj::String> getPythonRequirements(const Worker::Script::ModulesSource& source) {
```

Comment:
If all of this were moved into `api/pyodide.h/c++` could it be reused by both the internal repo and workerd? Or is the impl here specific to workerd

Reply:
Nope, it is workerd-specific because workerd has its own WorkerBundle definition.